rosbag2/test/rosbag2/rosbag2_test_fixture.hpp

```diff
@@ -112,72 +115,36 @@ class Rosbag2TestFixture : public Test
 
   void write_messages(
     const std::string & db_name,
-    std::vector<std::shared_ptr<rosbag2_storage::SerializedBagMessage>> messages)
+    std::vector<std::shared_ptr<rosbag2_storage::SerializedBagMessage>> messages,
+    std::map<std::string, std::string> topic_types)
   {
     rosbag2_storage::StorageFactory factory;
     auto storage = factory.open_read_write(db_name, "sqlite3");
+    for (auto topic_and_type : topic_types) {
+      storage->create_topic(topic_and_type.first, topic_and_type.second);
+    }
 
-    if (storage) {
-      storage->create_topic();
-      for (auto msg : messages) {
-        storage->write(msg);
-      }
+    for (auto msg : messages) {
+      storage->write(msg);
     }
   }
 
-  std::shared_ptr<rosbag2_storage::SerializedBagMessage> serialize_message(std::string message)
+  template<typename MessageT>
+  std::shared_ptr<rosbag2_storage::SerializedBagMessage> serialize_message(
+    std::string topic, typename MessageT::_data_type message)
```

Comment:
You are right. This is still used by the `write_integration_test` which is massively rewritten in #31. So I would prefer to address this in #31 to avoid an unecessary rebase.